src/Compilers/Core/Portable/DiagnosticAnalyzer/SuppressMessageAttributeState.cs

```diff
@@ -99,28 +125,30 @@ private bool IsDiagnosticSuppressed(string id, ISymbol symbol)
             {
                 var associated = ((IMethodSymbol)symbol).AssociatedSymbol;
                 if (associated != null &&
-                    (IsDiagnosticLocallySuppressed(id, associated) || IsDiagnosticGloballySuppressed(id, associated)))
+                    (IsDiagnosticLocallySuppressed(id, associated, out info) || IsDiagnosticGloballySuppressed(id, associated, out info)))
                 {
                     return true;
                 }
             }
 
-            if (IsDiagnosticLocallySuppressed(id, symbol) || IsDiagnosticGloballySuppressed(id, symbol))
+            if (IsDiagnosticLocallySuppressed(id, symbol, out info) || IsDiagnosticGloballySuppressed(id, symbol, out info))
             {
                 return true;
             }
 
             // Check for suppression on parent symbol
             var parent = symbol.ContainingSymbol;
-            return parent != null && IsDiagnosticSuppressed(id, parent);
+            return parent != null && IsDiagnosticSuppressed(id, parent, out info);
         }
 
-        private bool IsDiagnosticSuppressed(string id, Location location)
+        private bool IsDiagnosticSuppressed(string id, Location location, out SuppressMessageInfo info)
         {
             Debug.Assert(id != null);
             Debug.Assert(location != null);
 
-            if (IsDiagnosticGloballySuppressed(id, symbolOpt: null))
+            info = default(SuppressMessageInfo);
```

Comment:
minor: You probably only need to initialize info at the end in the return false case.